modules/_labs/button/react/lib/IconButton.tsx

```diff
@@ -0,0 +1,224 @@
+import * as React from 'react';
+import {colors, spacing, borderRadius} from '@workday/canvas-kit-react-core';
+import {focusRing} from '@workday/canvas-kit-react-common';
+import {SystemIcon} from '@workday/canvas-kit-react-icon';
+import {CanvasSystemIcon} from '@workday/design-assets-types';
+import {IconButtonVariant, ButtonSize, ButtonColors} from './types';
+import {ButtonContainer} from './parts';
+
+export interface IconButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  /**
+   * The type of the IconButton.
+   * @default IconButtonVariant.Circle
+   */
+  variant: IconButtonVariant;
+  /**
+   * The accessibility label to indicate the action triggered by clicking the IconButton.
+   */
+  'aria-label': string;
+  /**
+   * The size of the IconButton.
+   * @default ButtonSize.Medium
+   */
+  size: ButtonSize.Small | ButtonSize.Medium;
+  /**
+   * If true, toggle the IconButton on.
+   * @default false
+   */
+  toggled?: boolean;
+  /**
+   * The ref to the button that the styled component renders.
+   */
+  buttonRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * The icon of the IconButton.
+   */
+  icon?: CanvasSystemIcon;
+  /**
+   * The function called when the IconButton toggled state changes.
+   */
+  onToggleChange?: (toggled: boolean | undefined) => void;
+}
+
+const IconButton = (props: IconButtonProps) => {
+  const {buttonRef, size, variant, onToggleChange, icon, toggled, children, ...elemProps} = props;
+
+  const isInitialMount = React.useRef(true);
+
+  // Only call onToggleChange on update - not on first mount
+  React.useEffect(() => {
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
+    } else {
+      if (props.toggled && typeof props.onToggleChange === 'function') {
+        props.onToggleChange(toggled);
+      }
+    }
+  }, [props.toggled]);
```

Comment:
Would we need to add `props.onToggleChange` here too? We do use it in the effect. Dunno if it applies here but this is from the React docs:

> If you use this optimization, make sure the array includes all values from the component scope (such as props and state) that change over time and that are used by the effect.

Reply:
Nice catch 👍